perf_workflow/run_perf_suite/ccr_perf_test.py

```diff
@@ -0,0 +1,294 @@
+        root.setLevel(logging.INFO)
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(logging.INFO)
+        root.addHandler(handler)
+
+    def workload_options(self, leaderCluster, followerCluster):
+        telemetry_options = {
+            "ccr-stats-sample-interval": 1,
+            "ccr-stats-indices": {"follower": [self.test_config["Workload"]]},
+        }
+        workload_options = {
+            "telemetry-params": json.dumps(telemetry_options),
+            "workload-params": "number_of_shards:{},number_of_replicas:{}".format(
+                self.test_config.get("Shards", 1), self.test_config.get("Replica", 0)
+            ),
+        }
+        if self.security:
+            # TODO: Move to configurable auth username and pwd.
+            client_options = {
+                "default": {
+                    "use_ssl": True,
+                    "basic_auth_user": "admin",
+                    "basic_auth_password": "admin",
+                    "verify_certs": False,
+                },
+                "follower": {
+                    "use_ssl": True,
+                    "basic_auth_user": "admin",
+                    "basic_auth_password": "admin",
+                    "verify_certs": False,
+                },
+            }
+            workload_options["client-options"] = json.dumps(client_options)
+        return json.dumps(workload_options)
+
+    def setup_replication(self, leaderCluster, followerCluster, workload_name):
+        self.setup_seed_nodes(leaderCluster, followerCluster)
+        self.setup_autofollow(followerCluster, workload_name)
+
+    def setup_autofollow(self, followerCluster, workload_name):
+        url = "".join(
+            [followerCluster.endpoint_with_port(), "/_plugins/_replication/_autofollow"]
+        )
+        body = {
+            "leader_alias": self.CONNECTION_ALIAS,
+            "name": "all",
+            "pattern": workload_name,
+        }
+        if self.security:
+            body["use_roles"] = {
+                "leader_cluster_role": "all_access",
+                "follower_cluster_role": "all_access",
+            }
+
+        headers = {"Content-Type": "application/json"}
+        resp = requests.post(
+            url=url,
+            json=body,
+            headers=headers,
+            auth=HTTPBasicAuth("admin", "admin"),
+            verify=False,
+        )
+        if resp.status_code != 200:
+            raise RuntimeError(
+                "Unable to update trigger autofollow replication: {}".format(
+                    resp.reason
+                )
+            )
+
+    def setup_seed_nodes(self, leaderCluster, followerCluster):
+        requests.packages.urllib3.disable_warnings()
+        # Fetch seed nodes from leader cluster
+        url = "".join([leaderCluster.endpoint_with_port(), "/_cat/nodes?format=json"])
+        resp = requests.get(url=url, auth=HTTPBasicAuth("admin", "admin"), verify=False)
```

Comment:
Can you declare username/password and then use?

Reply:
Replaced with `self.CREDS`